File: exercise/php/day03/src/Preparation/Gift.php
<?php

declare(strict_types = 1);

namespace Preparation;

class Gift
{
	private string $name;

	private float $weight;

	private string $color;

	private string $material;

	private array $attributes = [];

	public function __construct(string $name, float $weight, string $color, string $material)
	{
		$this->name = $name;
		$this->weight = $weight;
		$this->color = $color;
		$this->material = $material;
	}

	public function __toString(): string
	{
		return "A {$this->color}-colored {$this->name} weighting {$this->weight} kg made in {$this->material}";
	}

	public function addAttribute(string $key, string $value): void
	{
		$this->attributes[$key] = $value;
	}

	public function recommendedAge(): int
	{
		return isset($this->attributes['recommendedAge']) && is_numeric($this->attributes['recommendedAge']) ?
			(int) $this->attributes['recommendedAge'] :
			0;
	}
}

File: exercise/C#/day03/Preparation/Gift.cs
namespace Preparation
{
    public class Gift(string name, double weight, string color, string material)
    {
        private readonly Dictionary<string, string> _attributes = new();
        public void AddAttribute(string key, string value) => _attributes[key] = value;
        public int RecommendedAge() => int.TryParse(_attributes["recommendedAge"], out var age) ? age : 0;
        public override string ToString() => $"A {color}-colored {name} weighting {weight} kg made in {material}";
    }
}